### PR TITLE
Added Configuration for SASS in nuxt

### PR DESCRIPTION
**Pull request by Andrew99xx**
-Downloaded dependencies
-Changed nuxt config file
- Changed index.vue for testing purpose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ You will need these tools to get started with dev work on this software:
 1. [Node.js](https://nodejs.org/en)
 2. [PNPM](https://pnpm.io/)
 3. [Task](https://taskfile.dev/)
-4. [Pre Commit](https://pre-commit.com/)
+4. [Pre-Commit](https://pre-commit.com/)
```
Comment (Sagar-Kap) [inline README.md:20]: Try to keep the PR only focused on one issue at a time. Even if it is a minor change, do it with a separate PR.

File: front-end/assets/css/main.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// assets/styles.scss
```
Comment (Sagar-Kap) [inline front-end/assets/css/main.scss:1]: Rename the `dir` as `styles` instead `css`.

File: front-end/assets/css/main.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+$primary-color: #3498db;
```
Comment (Jarmos-san) [inline front-end/assets/css/main.scss:2]: Refactor and move variables to a `_variables.scss` partial.

File: front-end/assets/css/main.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+body {
```
Comment (Jarmos-san) [inline front-end/assets/css/main.scss:4]: ?? 

Add CSS resets instead of adding arbitrary and experimental CSS code. @Sagar-Kap will share an example of CSS resets with you. Remember to add the CSS resets as a SASS partial (named as `_resets.scss`).

File: front-end/nuxt.config.ts [hunk 5]
```diff
@@ -3,4 +3,8 @@ export default defineNuxtConfig({
   compatibilityDate: "2024-04-03",
   devtools: { enabled: true },
   modules: ["@nuxt/eslint"],
+  css: [
```
Comment (Jarmos-san) [inline front-end/nuxt.config.ts:6]: Utilise SASS partials which should be imported to a `~/assets/styles/main.scss` file. Reconfigure Vite (as a build tool for Nuxt) to use that "root SASS module" to apply the global styles as follows:

```typescript
export default defineNuxtConfig({
  vite: {
    css: {
      preprocessorOptions: {
        scss: {
          additionalData: '@use "~/assets/styles/main.scss" as *;'
        }
      }
    }
  }
})
```

File: front-end/pages/index.vue [hunk 6]
```diff
@@ -1,7 +1,29 @@
...
+  <div class="container">
+    <h1>Welcome to Nuxt with SASS</h1>
+    <p>This is a demo page to test SASS integration in a Nuxt.js project.</p>
+  </div>
 </template>
 
-<script setup></script>
+<script>
+export default {
+  head() {
```
Comment (Jarmos-san) [inline front-end/pages/index.vue:10]: ??

Use appropriate publicly documented API instead. Check out the `useHead()` Composable instead (read the docs for more information).

File: front-end/pages/index.vue [hunk 7]
```diff
@@ -1,7 +1,29 @@
...
+    return {
+      title: "Nuxt SASS Demo",
+    };
+  },
+};
+</script>
 
-<style></style>
+<style lang="scss" scoped>
+@import "../assets/css/main.scss" .container {
```
Comment (Jarmos-san) [inline front-end/pages/index.vue:19]: Don't import arbitrary SASS modules to an SFC. The SFC should be idempotent, unless stated specifically with a valid reason.

Also is this even legal SASS syntax? Write clean and legible code, not something produced by ChatGPT.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+	"devDependencies": {
```
Comment (Jarmos-san) [inline package.json:2]: Webpack? Don't use ChatGPT blindly, read the official docs instead (see [official instructions](https://nuxt.com/docs/getting-started/styling#using-preprocessors) on setting up SASS).

**Review by Jarmos-san — CHANGES_REQUESTED** [review]
Besides, the requested changes, take care of the following concerns as well:

1. Follow a standard practice of writing well defined PR titles (the CommitLint specification is recommended).
2. Write clean and legible code.
3. Don't copy + paste arbitrary code from ChatGPT. If you need advice or suggestion, discuss through the comment section before implementing a piece of code. This will save everyone's time and effort.

File: front-end/assets/css/main.scss (hunk 3 above)
Comment (Andrew99xx) [reply front-end/assets/css/main.scss:2]: thank you for helping me out. this was done because i wanted to check whether SASS is working as expected or ot. that is only draft ones to quickly check

File: package.json (hunk 8 above)
Comment (Andrew99xx) [reply package.json:2]: I used youtube videos and documentations, not chatgpt

Comment (Jarmos-san) [reply package.json:2]: Read the official docs instead or ask for guidance on the relevant issue thread instead.

File: front-end/assets/css/main.scss (hunk 3 above)
Comment (Jarmos-san) [reply front-end/assets/css/main.scss:2]: You can experiment and check whether SASS is working or not while also writing clean code at the same time.

**Comment by Jarmos-san** [comment]
This PR is superseded by #169 and should be closed since its messy and is blocking work on the rest of the project. @Andrew99xx take notes from the linked PR on how to write good, clean and reviewable PRs.
